### PR TITLE
Phase 3/develop

### PR DESCRIPTION
**Pull request by andybit-okutama**

### CODE REVIEW

**Comment by ericgpks** [comment]
Please fix this file and this description to "m_corporate_personalities" because the difference of this description and the second picture's description causes the error.
image

image

**Comment by ericgpks** [comment]
In admin/companies_conroller.rb, please change the end_date to one day before of one month later of start_date. Contract_end_date is same.
image

**Comment by ericgpks** [comment]
In admin/companies_conroller.rb, different data is set to billing between create and update. Thus, billing data comes to be wrong when I update company data. Please check it.

**Comment by ericgpks** [comment]
You need to define the size of picture in service area. If you don't do it, it looks like this depending on the uploaded file.
screencapture-192-168-33-13-3000-2019-03-23-17_39_00

**Comment by ericgpks** [comment]
In "doc-dnus\db\fixtures\07_service.rb", these category_id are wrong. Please check the following sheet and fix them.
https://docs.google.com/spreadsheets/d/1RbWjBPBzOYvyQ_mb6bf612aQAe8urVXTxFBqnQ1uHys/edit#gid=763265674
image

**Comment by ericgpks** [comment]
Is it only my circumstance that has an extra space between checkbox and image?
スクリーンショット (747)

File: app/assets/stylesheets/object/user/authentication/signup.scss [hunk 1]
```diff
@@ -43,6 +43,27 @@
   }
 }
 
+@media screen and (min-width: 769px){
+  .p-in-register{
+    width: 600px;
+    &__form{
+      background: #FFFFFF;
```
Comment (ericgpks) [inline app/assets/stylesheets/object/user/authentication/signup.scss]: 💯

File: app/assets/stylesheets/object/user/businesses/show.scss [hunk 2]
```diff
@@ -0,0 +1,87 @@
+// Place all the styles related to the companies controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
+.p-i-businesses{
```
Comment (ericgpks) [inline app/assets/stylesheets/object/user/businesses/show.scss]: 💯

File: app/assets/stylesheets/object/admin/articles/form.scss [hunk 3]
```diff
@@ -0,0 +1,95 @@
+.p-i-add-articles{
+  margin-top: 15px;
+  .btn-delete {
+    color: #D0021B;
+    text-decoration: underline;
+  }
+  .btn-dropdown {
+    color: #fff;
```
Comment (ericgpks) [inline app/assets/stylesheets/object/admin/articles/form.scss]: Please write contents in order from A to Z all in this file.

File: app/controllers/admin/articles_controller.rb [hunk 4]
```diff
@@ -6,31 +6,112 @@ def index
...
+      ArticleService.where(article_id: @article.id).delete_all
+      @services = params['article']['article_service'] # Setting Service for array
+      if !@services.nil?
+          @services.each do |service|  #looping service
+          @service_type = MService.find(service) #getting plan
+          @article_service = ArticleService.create(article_id: @article.id, service_id: @service_type.id) ## create database
+ 
+        end
+      end
+      redirect_to admin_articles_path
```
Comment (ericgpks) [inline app/controllers/admin/articles_controller.rb]: ❤️

File: app/helpers/admins/articles_helper.rb [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  end
+  def setstatus(status)
+    if status == 1
+      return "公開中"
+    else
+      if status == 2
+        return "限定公開"
+      else
+     	return "下書き"
+	  end
```
Comment (ericgpks) [inline app/helpers/admins/articles_helper.rb]: Please check this position.

File: app/models/m_service.rb [hunk 6]
```diff
@@ -1,4 +1,6 @@
 class MService < ApplicationRecord
   belongs_to :m_service_category, foreign_key: 'category_id'
   has_many :business_services
+  has_many :ArticleService
```
Comment (ericgpks) [inline app/models/m_service.rb]: It must be multiple when you use "has_many".

File: app/views/admin/layout/_nav.html.haml [hunk 7]
```diff
@@ -2,4 +2,4 @@
     = link_to "ダッシュボード", "#"
     = link_to "企業", admin_companies_path
     = link_to "資料", admin_businesses_path
-    = link_to "記事", "#", :style => "border-bottom: 1px solid #FFFFFF;"
\ No newline at end of file
+    = link_to "記事", admin_articles_path
```
Comment (ericgpks) [inline app/views/admin/layout/_nav.html.haml]: Will you change these links?

File: app/assets/stylesheets/object/user/businesses/show.scss [hunk 8]
```diff
@@ -46,6 +46,25 @@
     margin-bottom: 12px;
   }
 
+  .btn-green{
```
Comment (ericgpks) [inline app/assets/stylesheets/object/user/businesses/show.scss]: 💯

File: app/views/users/layout/_footer.html.haml [hunk 9]
```diff
@@ -4,27 +4,27 @@
       %h5.p-users-footer__title dnus(ディヌス)について
       %ul.list-unstyled
         %li
-          %a{:href => "#!"} ユーザ向け利用規約
+          %a{:href => "/terms"} ユーザ向け利用規約
```
Comment (ericgpks) [inline app/views/users/layout/_footer.html.haml]: You can use "link_to" when you write %a.
Please reference this.
https://api.rubyonrails.org/v5.2.2/classes/ActionView/Helpers/UrlHelper.html#method-i-link_to

File: app/assets/stylesheets/object/user/article/article.scss [hunk 10]
```diff
@@ -0,0 +1,189 @@
+// Place all the styles related to the service controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
+
+.p-i-article{
```
Comment (ericgpks) [inline app/assets/stylesheets/object/user/article/article.scss]: Please write contents in order from A to Z all in this file.

File: app/controllers/users/articles_controller.rb [hunk 11]
```diff
@@ -1,7 +1,12 @@
 class Users::ArticlesController < ApplicationController
-  before_action :authenticate_user!
+  PER = 2
   def index
-    @articles = Article.all
+  	@PER = PER
+    @articles = Article.where('publicity_flag = 1').order(created_at: "DESC").all
```
Comment (ericgpks) [inline app/controllers/users/articles_controller.rb]: ❤️

File: app/assets/stylesheets/object/user/service/form.scss [hunk 12]
```diff
@@ -139,6 +139,24 @@
     box-shadow: 5px 5px 4px 0px rgba(0, 0, 0, 0.4)
   
   }
+  &__grey {
```
Comment (ericgpks) [inline app/assets/stylesheets/object/user/service/form.scss]: ❤️

File: app/views/elements/service/_businessdata.html.haml [hunk 13]
```diff
@@ -35,12 +35,12 @@
                     #{service.name}
             .table-striped.p-i-service__table-des
               = business.explanation
-            .table-striped.p-i-service__table-btn
-              = link_to "詳細を見る", business_path(id: business.id), controller: "businesses", method: :get, class: "btn p-i-service__green"
+            .table-striped.p-i-service__table-btn.row
+              = link_to "詳細を見る", business_path(id: business.id), controller: "businesses", method: :get, class: "btn p-i-service__green col-md-5"
```
Comment (ericgpks) [inline app/views/elements/service/_businessdata.html.haml]: 💯

File: app/views/users/layout/_header.html.haml [hunk 14]
```diff
@@ -12,8 +12,8 @@
       %a.p-users-navbar.icon{href: "javascript:void(0);", onclick: "nav()"}
         %i.fa.fa-bars
       %a{href: root_path} ホーム
-      %a{href: "#"} カテゴリーから探す
-      %a{href: "#"} ニュース・記事
+      %a{href: categories_path } カテゴリーから探す
```
Comment (ericgpks) [inline app/views/users/layout/_header.html.haml]: You can use "link_to" when you want to write %a.
Please reference this article.
https://api.rubyonrails.org/v5.2.2/classes/ActionView/Helpers/UrlHelper.html#method-i-link_to

Comment (andybit-okutama) [reply app/views/users/layout/_header.html.haml]: Fixed

File: app/assets/stylesheets/object/user/article/article.scss (hunk 10 above)
Comment (andybit-okutama) [reply app/assets/stylesheets/object/user/article/article.scss]: Fix

File: app/assets/stylesheets/object/admin/articles/form.scss (hunk 3 above)
Comment (andybit-okutama) [reply app/assets/stylesheets/object/admin/articles/form.scss]: fix

**Comment by andybit-okutama** [comment]
Can i merge this and make a new pull request.
i have update the phase3 with newest code

File: app/controllers/users/download_controller.rb [hunk 15]
```diff
@@ -1,2 +1,35 @@
...
+          zos.write(buffer)          
+      end
+    end
+    # Zip::File.open(t.path, Zip::File::Write) do |zipfile|
+    #   @businesses.each do |attachment|
+    #     zipfile.add(attachment.document.file.filename, "#{Rails.root}/public#{attachment.document_url}")
+    #   end
+    # end
+    date = Time.now.strftime("%Y%d%m")
+    send_file t.path, :type => 'application/zip', :disposition => 'attachment', :filename => "dnus_documents_#{date}.zip"
```
Comment (ericgpks) [inline app/controllers/users/download_controller.rb]: Can you change this to the following style?
type: 'application/zip', disposition: 'attachment', filename: "dnus_documents_#{data}.zip"

**Comment by ericgpks** [comment]
Please fix this one and check if it works correctly.
If it is OK, please merge.

File: app/controllers/users/download_controller.rb (hunk 15 above)
Comment (andybit-okutama) [reply app/controllers/users/download_controller.rb]: date = Time.now.strftime("%Y%d%m") -> data = Time.now.strftime("%Y%d%m") 
this also right

Comment (ericgpks) [reply app/controllers/users/download_controller.rb]: Sorry. It doesn't need to change. It was just a mistyping.

File: app/assets/stylesheets/object/user/service/form.scss [hunk 16]
```diff
@@ -17,6 +17,28 @@
 	font-family:"ヒラギノ角ゴ Pro W3", "Hiragino Kaku Gothic Pro",Osaka, "メイリオ", Meiryo, "ＭＳ Ｐゴシック", "MS PGothic", sans-serif;
   padding:60px;
   }
+  &__bodyfix {
+    background: rgba(0,0,0,0.51);
```
[the remark this answers is not in the document]
Comment (andybit-okutama) [reply app/assets/stylesheets/object/user/service/form.scss]: Fix

File: app/models/m_service.rb (hunk 6 above)
Comment (andybit-okutama) [reply app/models/m_service.rb]: Will check

File: app/views/admin/layout/_nav.html.haml (hunk 7 above)
Comment (andybit-okutama) [reply app/views/admin/layout/_nav.html.haml]: Change

**Pull request merged**